lib/regexp/regexp.go

```diff
@@ -0,0 +1,258 @@
+// in starklark. required by starlark.HasAttrs interface
+func (r *Regex) Attr(name string) (starlark.Value, error) {
+	return builtinAttr(r, name, regexMethods)
+}
+
+var regexMethods = map[string]*starlark.Builtin{
+	"find":            starlark.NewBuiltin("find", find),
+	"find_all":        starlark.NewBuiltin("find_all", findAll),
+	"find_submatches": starlark.NewBuiltin("find_submatches", findSubmatches),
+	"matches":         starlark.NewBuiltin("matches", matches),
+	"replace_all":     starlark.NewBuiltin("replace_all", replaceAll),
+	"split":           starlark.NewBuiltin("split", split),
+}
+
+// AttrNames lists available dot expression strings for time. required by
+// starlark.HasAttrs interface
+func (r *Regex) AttrNames() []string { return builtinAttrNames(regexMethods) }
+
+func matches(_ *starlark.Thread, b *starlark.Builtin, args starlark.Tuple, kwargs []starlark.Tuple) (starlark.Value, error) {
+	var (
+		src string
+	)
+
+	if err := starlark.UnpackPositionalArgs(b.Name(), args, kwargs, 1, &src); err != nil {
+		return starlark.None, err
+	}
+
+	re := b.Receiver().(*Regex).re
+	return starlark.Bool(re.MatchString(src)), nil
+}
+
+func find(_ *starlark.Thread, b *starlark.Builtin, args starlark.Tuple, kwargs []starlark.Tuple) (starlark.Value, error) {
+	var (
+		src string
+	)
+
+	if err := starlark.UnpackPositionalArgs(b.Name(), args, kwargs, 1, &src); err != nil {
+		return starlark.None, err
+	}
+
+	re := b.Receiver().(*Regex).re
+	return starlark.String(re.FindString(src)), nil
+}
+
+func findAll(_ *starlark.Thread, b *starlark.Builtin, args starlark.Tuple, kwargs []starlark.Tuple) (starlark.Value, error) {
+	var (
+		src string
+		max int = -1
+	)
+
+	if err := starlark.UnpackPositionalArgs(b.Name(), args, kwargs, 1, &src, &max); err != nil {
+		return starlark.None, err
+	}
+
+	re := b.Receiver().(*Regex).re
+	return toTuple(re.FindAllString(src, max)), nil
+}
+
+func findSubmatches(_ *starlark.Thread, b *starlark.Builtin, args starlark.Tuple, kwargs []starlark.Tuple) (starlark.Value, error) {
+	var (
+		src string
+	)
+
+	if err := starlark.UnpackPositionalArgs(b.Name(), args, kwargs, 1, &src); err != nil {
+		return starlark.None, err
+	}
+
+	re := b.Receiver().(*Regex).re
+	return toTuple(re.FindStringSubmatch(src)), nil
+}
+
+func replaceAll(thread *starlark.Thread, b *starlark.Builtin, args starlark.Tuple, kwargs []starlark.Tuple) (starlark.Value, error) {
+	var (
+		src  string
+		repl starlark.Value
+	)
+
+	if err := starlark.UnpackPositionalArgs(b.Name(), args, kwargs, 2, &src, &repl); err != nil {
+		return starlark.None, err
+	}
+
+	re := b.Receiver().(*Regex).re
+	switch x := repl.(type) {
+	case starlark.Callable:
+		fn := func(matched string) string {
+			res, err := starlark.Call(thread, repl, starlark.Tuple{starlark.String(matched)}, nil)
+			if err != nil {
+				log.Printf("An error occured while executing the function: %s", err.Error())
```

Comment:
In this case, it's important we return the original err to the caller, so that the stack is preserved and reported to the user. No need for log.Print. Please add a comment and a test for this.